src/hooks/Orderbook/useDrawOrderbook.ts

```diff
@@ -0,0 +1,386 @@
+  }: {
+    barType: 'depth' | 'size';
+    ctx: CanvasRenderingContext2D;
+    depthOrSizeValue: number;
+    gradientMultiplier: number;
+    histogramAccentColor: string;
+    histogramSide: 'left' | 'right';
+    idx: number;
+  }) => {
+    const { x1, x2, y1, y2 } = getRektFromIdx({
+      idx,
+      rowHeight,
+      canvasWidth,
+      canvasHeight,
+      side,
+    });
+
+    // X values
+    const maxHistogramBarWidth = x2 - x1 - (barType === 'size' ? 8 : 2);
+    const barWidth = depthOrSizeValue
+      ? Math.min((depthOrSizeValue / histogramRange) * maxHistogramBarWidth, maxHistogramBarWidth)
+      : 0;
+
+    const { gradient, bar } = getHistogramXValues({
+      barWidth,
+      canvasWidth,
+      gradientMultiplier,
+      histogramSide,
+    });
+
+    // Gradient
+    let linearGradient;
+
+    try {
+      linearGradient = ctx.createLinearGradient(gradient.x1, y1, gradient.x2, y2);
+      linearGradient.addColorStop(0, histogramAccentColor);
+      linearGradient.addColorStop(1, 'rgba(0, 0, 0, 0)');
+      ctx.fillStyle = linearGradient;
+    } catch (err) {
+      ctx.fillStyle = 'rgba(0, 0, 0, 0)';
+    }
+
+    ctx.beginPath();
+
+    // Bar
+    const { bar: y } = getYForElements({ y: y1, rowHeight });
+
+    if (ctx.roundRect) {
+      ctx.roundRect?.(
+        bar.x1,
+        y,
+        bar.x2,
+        rowHeight - 2,
+        histogramSide === 'right' ? [2, 0, 0, 2] : [0, 2, 2, 0]
+      );
+    } else {
+      ctx.rect(bar.x1, y, bar.x2, rowHeight - 2);
+    }
+
+    ctx.fill();
+  };
+
+  const drawText = ({
+    animationType = OrderbookRowAnimationType.NONE,
+    ctx,
+    idx,
+    mine,
+    price,
+    size,
+  }: {
+    animationType?: OrderbookRowAnimationType;
+    ctx: CanvasRenderingContext2D;
+    idx: number;
+    mine?: number;
+    price?: number;
+    size?: number;
+  }) => {
+    const { y1 } = getRektFromIdx({
+      idx,
+      rowHeight,
+      canvasWidth,
+      canvasHeight,
+      side,
+    });
+
+    const { text: y } = getYForElements({ y: y1, rowHeight });
+
+    let textColor: string;
+    let updatedTextColor: string | undefined;
+
+    switch (animationType) {
+      case OrderbookRowAnimationType.REMOVE: {
+        textColor = color.text1;
+        break;
+      }
+
+      case OrderbookRowAnimationType.NEW: {
+        updatedTextColor = side === 'bid' ? color.positive : color.negative;
+        textColor = color.text2;
+        break;
+      }
+
+      default: {
+        textColor = color.text2;
+        break;
+      }
+    }
+
+    // Size text
+    if (size) {
+      ctx.fillStyle = updatedTextColor ?? textColor;
+      ctx.fillText(
+        MustBigNumber(size).toFixed(stepSizeDecimals ?? TOKEN_DECIMALS),
+        getXByColumn({ canvasWidth, colIdx: 0 }) - ROW_PADDING_RIGHT,
+        y
+      );
+    }
+
+    // Price text
+    if (price) {
+      ctx.fillStyle = textColor;
+      ctx.fillText(
+        MustBigNumber(price).toFixed(tickSizeDecimals ?? SMALL_USD_DECIMALS),
```

Comment:
hm unsure if I'm missing something, but isn't `tickSizeDecimals` (and `stepSizeDecimals`) on line 246 [guaranteed](https://github.com/dydxprotocol/v4-web/pull/227/files#diff-baa2ed13ef2f5329060a1d669ed4ac62732644d4817aad4821b2f1595d15460bR51) to be non-null?

Reply:
<img width="420" alt="Screen Shot 2024-01-22 at 1 09 37 PM" src="https://github.com/dydxprotocol/v4-web/assets/13111994/23177896-c5a6-47ac-b83e-12452a11a289">

typescript still infers `null` as a possible value

Reply:
the `= 2` only handles the case when `stepSizeDecimals` and `tickSizeDecimals` are undefined. Unfortunately they are both `Nullable<number>` (`null | undefined | number`)